build.sh

```diff
@@ -1,12 +1,42 @@
 #!/bin/bash
-set -eo pipefail
+set -eox pipefail
 
-if [ -z "$1" ] ; then
-  echo "Usage: ./build.sh \$VERSION"
+if [ -z "$1" ]; then
+  echo "Usage: ./build.sh \$VERSION_SHORT"
 else
-  VERSION="$1"
+  VERSION_SHORT="$1"
 fi
 
-cd "$VERSION" || exit
+VERSION=$(grep -oP '[0-9]+\.[0-9]+\.[0-9]+' "$VERSION_SHORT/Dockerfile" | head -1)
+DOCKER_REPO=factoriotools/docker_factorio_server
+cd "$VERSION_SHORT" || exit
 
-docker build . -t "factoriotools/docker_factorio_server:$VERSION"
+if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
+  if [ "$TRAVIS_BRANCH" == "master" ]; then
+    TAG="$VERSION -t $DOCKER_REPO:$VERSION_SHORT"
+  else
+    TAG="$TRAVIS_BRANCH"
+  fi
+
+  if [ -n "$EXTRA_TAG" ]; then
+    TAG="$TAG -t $DOCKER_REPO:$EXTRA_TAG"
+  fi
+fi
+
+# shellcheck disable=SC2086
+docker build . -t $DOCKER_REPO:$TAG
+
+docker images
+
+if [ "dirname $(git diff --name-only HEAD^) | head -1" == "$VERSION" ]; then
+  # echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+  docker push "$DOCKER_REPO:latest"
+
+  if [ -n "$EXTRA_TAG" ]; then
+    docker push "$DOCKER_REPO:$EXTRA_TAG"
+  fi
+
+  if [ -n "$TRAVIS_TAG" ]; then
+    docker push "$DOCKER_REPO:$VERSION"
+  fi
```

Comment:
What about `docker push` for version short?

Reply:
Missed that one. Nice catch.